codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocol.kt

```diff
@@ -92,6 +97,37 @@ interface ServerProtocol : Protocol {
             .toType().resolve("protocol::$protocolModulePath::runtime_error::RuntimeError")
 }
 
+class ServerBuilderInstantiator(private val symbolProvider: RustSymbolProvider, private val symbolParseFn: (Shape) -> ReturnSymbolToParse) :
```

Comment:
This class should not be in this file. There's nothing about builders that is protocol-aware.

Also, we shouldn't rely on `ClientBuilderInstantiator` here. `BuilderGenerator.kt` (which hosts `ClientBuilderInstantiator`) is currently in `codegen-core`, but it shouldn't (see https://github.com/awslabs/smithy-rs/issues/1970), it should be in `codegen-client`. I'm ok merging this PR before #1970 gets tackled, but we shouldn't add _more_ tech debt.